great_expectations/rule_based_profiler/types/parameter_container.py

```diff
@@ -44,7 +44,7 @@
     )
     ^ (
         Suppress(Literal("["))
-        + Word(nums).setParseAction(lambda s, l, t: [int(t[0])])
```

Comment:
@cdkini This gives us subscripting grammar for dictionaries, lists, including negative offsets.

Reply:
But why the change?

Reply:
@cdkini Because in the original implementation, I forgot to add the ability for users to specify things like: `max_value: $parameter.my_max_event_ts.value[-1]`